packages/@lwc/engine/src/__tests__/events.spec.ts

```diff
@@ -182,8 +182,8 @@ describe('events', () => {
             divWithClickHandler.addEventListener('click', function(event) {
                 target = event.target;
             });
-            divWithClickHandler.appendChild(elm);
             document.body.appendChild(divWithClickHandler);
+            divWithClickHandler.appendChild(elm);
```

Comment:
It is because the custom element will only be hydrated when it is attached to a connected parent. 

tl;dr;
This was an optimization in node-reaction for appendChild() and insertBefore(). The lib will only trigger the connected callback if the appended child is a qualifying node and the parent is connected.